pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/TransformFunctionFactory.java

```diff
@@ -112,13 +115,24 @@ public static TransformFunction get(TransformExpressionTree expression, Map<Stri
     switch (expression.getExpressionType()) {
       case FUNCTION:
         String functionName = expression.getValue();
-        Class<? extends TransformFunction> transformFunctionClass = TRANSFORM_FUNCTION_MAP.get(functionName);
+        Class<? extends TransformFunction> transformFunctionClass;
+        FunctionInfo functionInfo = null;
+        if (FunctionRegistry.containsFunctionByName(functionName)) {
+          transformFunctionClass = ScalarTransformFunctionWrapper.class;
+          functionInfo = FunctionRegistry.getFunctionByName(functionName);
+        } else {
+          transformFunctionClass = TRANSFORM_FUNCTION_MAP.get(functionName);
+        }
+
         if (transformFunctionClass == null) {
           throw new BadQueryRequestException("Unsupported transform function: " + functionName);
         }
         try {
           transformFunction = transformFunctionClass.newInstance();
-        } catch (InstantiationException | IllegalAccessException e) {
+          if (functionInfo != null) {
+            ((ScalarTransformFunctionWrapper) transformFunction).setFunction(functionName, functionInfo);
+          }
+        } catch (Exception e) {
```

Comment:
Don't catch all Exceptions because it will also catch the BadQueryRequestException. We treat BadQueryRequestException differently on the caller side

Reply:
Why will a constructor throw BadQueryRequestException since this catch block is meant for only the constructor?

Reply:
Never mind, I thought this try-catch is around line 128